zebrad/src/components/inbound.rs

```diff
@@ -155,9 +155,20 @@ impl Service<zn::Request> for Inbound {
                 debug!("ignoring unimplemented request");
                 async { Ok(zn::Response::Nil) }.boxed()
             }
-            zn::Request::FindBlocks { .. } => {
-                debug!("ignoring unimplemented request");
-                async { Ok(zn::Response::Nil) }.boxed()
+            zn::Request::FindBlocks { known_blocks, stop } => {
+                let request = zs::Request::FindBlockHashes { known_blocks, stop };
+                self.state.call(request).map_ok(|resp| match resp {
+                        zs::Response::BlockHashes(hashes) => {
+                            if hashes.is_empty() {
+                                zn::Response::Nil
+                            }
+                            else {
+                                zn::Response::BlockHashes(hashes)
+                            }
+                        },
```

Comment:
We try to avoid multi-level matches, because they're harder to read:
```suggestion
                        zs::Response::BlockHashes(hashes) if hashes.is_empty() => zn::Response::Nil,
                        zs::Response::BlockHashes(hashes) => zn::Response::BlockHashes(hashes),
```